homeassistant/components/onvif/parsers.py

```diff
@@ -353,16 +362,38 @@ async def async_parse_last_reset(uid: str, msg) -> Event:
 
     Topic: tns1:Monitoring/OperatingTime/LastReset
     """
+    date_time = datetime_or_zero(msg.Message._value_1.Data.SimpleItem[0].Value)
     try:
         return Event(
             f"{uid}_{msg.Topic._value_1}",
             "Last Reset",
             "sensor",
             "timestamp",
             None,
-            dt_util.as_local(
-                dt_util.parse_datetime(msg.Message._value_1.Data.SimpleItem[0].Value)
-            ),
+            dt_util.as_local(date_time),
+            entity_enabled=False,
+        )
+    except (AttributeError, KeyError, ValueError):
```

Comment:
What raises these errors?

Reply:
`AttributeError` and `KeyError` are potentially thrown when accessing `msg.Message._value_1.Data.SimpleItem[0].Value`. `ValueError` is potentially thrown when attempting to parse the date.

Reply:
Do you mean parse date with `dt_util.as_local`?

We should be passing a valid datetime instance to `dt_util.as_local`. If the date is invalid we should return early before trying to modify it further.

Reply:
No modified so that datetime_or_zero can only return a valid datetime instance.

Reply:
Can we remove `ValueError` here then?

Reply:
@MartinHjelmare probably yes. There are no tests for this code though so I want to be careful.

@hunterjm ok to remove ValueError?

Reply:
Sure. As long as `datetime_or_zero` is used in all events parsing dates, let's remove it from all of the parsers.